build/Build.cs

```diff
@@ -0,0 +1,616 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Nuke.Common;
+using Nuke.Common.CI.TeamCity;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Tools.AzureSignTool;
+using Nuke.Common.Tools.DotNet;
+using Nuke.Common.Tools.NuGet;
+using Nuke.Common.Tools.OctoVersion;
+using Nuke.Common.Tools.SignTool;
+using Nuke.Common.Utilities.Collections;
+using Serilog;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+using static Nuke.Common.Tools.Git.GitTasks;
+using static Nuke.Common.Tools.AzureSignTool.AzureSignToolTasks;
+
+namespace Calamari.Build
+{
+    class Build : NukeBuild
+    {
+        const string LinuxRuntime = "linux-x64";
+        const string WindowsRuntime = "win-x64";
+        const string CiBranchNameEnvVariable = "OCTOVERSION_CurrentBranch";
+        const string RootProjectName = "Calamari";
+
+        readonly string[] TimestampUrls =
+        {
+            "http://timestamp.digicert.com?alg=sha256",
+            "http://timestamp.comodoca.com"
+        };
+
+        [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+        readonly Configuration Configuration =
+            IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+        [Solution] [Required] readonly Solution? Solution;
+
+        [Parameter("Whether to auto-detect the branch name - this is okay for a local build, "
+                   + "but should not be used under CI.")]
+        readonly bool AutoDetectBranch = IsLocalBuild;
+
+        [Parameter("Branch name for Calamari to use to calculate the version number. Can be set "
+                   + "via the environment variable " + CiBranchNameEnvVariable + ".", Name = CiBranchNameEnvVariable)]
+        readonly string? BranchName;
+
+        [Parameter("Run packing step in parallel")] 
+        readonly bool PackInParallel;
+
+        [Parameter("Build verbosity")] 
+        readonly string BuildVerbosity = "Minimal";
+
+        [Parameter("Sign Binaries")] 
+        readonly bool SignBinaries;
+
+        [Parameter("Append Timestamp")] 
+        readonly bool AppendTimestamp;
+
+        [Parameter("Set Calamari Version on OctopusServer")] 
+        readonly bool SetOctopusServerVersion;
+
+        [Parameter("AzureKeyVaultUrl")] 
+        readonly string? AzureKeyVaultUrl;
+
+        [Parameter("AzureKeyVaultAppId")] 
+        readonly string? AzureKeyVaultAppId;
+
+        [Parameter("AzureKeyVaultAppSecret")] 
+        readonly string? AzureKeyVaultAppSecret;
```

Comment:
Probably should be `[Secret]` too - like server: https://github.com/OctopusDeploy/OctopusDeploy/blob/master/nuke-build/Build.cs#L76

Reply:
Ah yep good catch.

Reply:
Done.